src/interactive-window/interactiveWindow.ts

```diff
@@ -160,6 +162,8 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         } else if (this.isWebExtension) {
             this.insertInfoMessage(DataScience.noKernelsSpecifyRemote()).ignoreErrors();
         }
+
+        this.codeGenerator = codeGeneratorFactory.getOrCreate(this.notebookDocument);
```

Comment:
If we're going to own this object (by storing a ref here), then we should dispose this object in the `dispose` method of this class.